pkg/collector/smi_collector.go

```diff
@@ -0,0 +1,109 @@
+package collector
+
+import (
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// SmiCollector defines an Smi Collector struct
+type SmiCollector struct {
+	BaseCollector
+}
+
+var _ interfaces.Collector = &SmiCollector{}
+
+// NewSmiCollector is a constructor
+func NewSmiCollector(exporter interfaces.Exporter) *SmiCollector {
+	return &SmiCollector{
+		BaseCollector: BaseCollector{
+			collectorType: Smi,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *SmiCollector) Collect() error {
+	// Get all CustomResourceDefinitions in the cluster
+	allCrdsList, err := utils.GetResourceList([]string{"get", "crds", "-o", "jsonpath={..metadata.name}"}, " ")
+	if err != nil {
+		return err
+	}
+
+	// Directory where logs will be written to
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	// Filter to obtain a list of Smi CustomResourceDefinitions in the cluster
+	crdNameContainsSmiPredicate := func(s string) bool { return strings.Contains(s, "smi-spec.io") }
+	smiCrdsList := filter(allCrdsList, crdNameContainsSmiPredicate)
+	if len(smiCrdsList) == 0 {
+		// TODO should this return an error?
+		log.Printf("Cluster does not contain any SMI CustomResourceDefinitions\n")
+		return nil
+	}
+
+	collectSmiCrdDefinitions(collector, filepath.Join(rootPath, "smi_crd_definitions"), smiCrdsList)
```

Comment:
nit: crdDefinitions = custom resource definition Definitions
```suggestion
	collectSmiCrds(collector, filepath.Join(rootPath, "smi_crd_definitions"), smiCrdsList)
```

Reply:
thanks! :)

Reply:
Resolved in eeb2fdb